### PR TITLE
Fix related external link bug

### PR DESCRIPTION
**Pull request by gpeng**
When updating an artefact related external links were being 're-saved' with each update effectively doubling them each time. Destroying them was also broken. This PR fixes the bug and adds some tests.

[Zendesk Ticket](https://govuk.zendesk.com/agent/tickets/1316142)

### CODE REVIEW

File: app/controllers/artefacts_controller.rb [hunk 1]
```diff
@@ -210,7 +210,7 @@ def extract_parameters(params)
         "sections" => [],
         "specialist_sectors" => [],
         "related_artefact_slugs" => [],
-        "external_links_attributes" => [:title, :url, :_id, :_destroy],
+        "external_links_attributes" => [:title, :url, :id, :_destroy],
```
Comment (Davidslv) [inline app/controllers/artefacts_controller.rb]: What's the reason behind this change?

Comment (gpeng) [inline app/controllers/artefacts_controller.rb]: This array is used to permit the params from the form. The rails `fields_for` helper adds a hidden input with the name `id` when the the object, in this case an `ArtefactExternalLink` is rendered to the page. This field (along with `_destroy = true | false` gets submitted back with the form and is used by `accepts_nested_attributes_for` to determine what action to take. So... Previously, the `id` parameter was being filtered out resulting in every form submission creating a new `ArtefactExternalLink` (or loads of them). If you add an external link on an artefact on integration and then keep saving the page you'll see them 'multiply'.

**Comment by benhyland** [comment]
Graham has kindly begun my induction into the hallowed halls of rails users by explaining in detail how this works - the only puzzle piece missing is Mongoid's problem.
As a result I think this is good to go.
